numpy/lib/function_base.py

```diff
@@ -305,11 +305,16 @@ def average(a, axis=None, weights=None, returned=False):
     Returns
     -------
     average, [sum_of_weights] : array_type or double
-        Return the average along the specified axis. When returned is `True`,
+        Return the average along the specified axis. When `returned` is `True`,
         return a tuple with the average as the first element and the sum
-        of the weights as the second element. The return type is `Float`
-        if `a` is of integer type, otherwise it is of the same type as `a`.
-        `sum_of_weights` is of the same type as `average`.
+        of the weights as the second element. `sum_of_weights` is of the
+        same type as `average`. The result dtype follows a genereal pattern.
```

Comment:
since `average` is rendered as a link to the function and not the returned array, please replace this occurance of `average` with `retval` and also change the returned array from average to retval (line 307).